code/iaas/lb-instance/src/main/java/io/cattle/platform/lb/instance/dao/impl/LoadBalancerInstanceDaoImpl.java

```diff
@@ -0,0 +1,51 @@
+package io.cattle.platform.lb.instance.dao.impl;
+
+import static io.cattle.platform.core.model.tables.NetworkTable.NETWORK;
+import io.cattle.platform.archaius.util.ArchaiusUtil;
+import io.cattle.platform.core.dao.GenericMapDao;
+import io.cattle.platform.core.model.LoadBalancer;
+import io.cattle.platform.core.model.LoadBalancerHostMap;
+import io.cattle.platform.core.model.Network;
+import io.cattle.platform.db.jooq.dao.impl.AbstractJooqDao;
+import io.cattle.platform.lb.instance.dao.LoadBalancerInstanceDao;
+import io.cattle.platform.object.ObjectManager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import com.netflix.config.DynamicStringProperty;
+
+public class LoadBalancerInstanceDaoImpl extends AbstractJooqDao implements LoadBalancerInstanceDao {
+
+    private static final DynamicStringProperty MANAGED_NETWORK_NAME = ArchaiusUtil
+            .getString("lb.instance.network.name");
+
+    GenericMapDao mapDao;
+    ObjectManager objectManager;
+
+    @Override
+    public List<Long> getLoadBalancerHosts(long lbId) {
+        List<Long> hostIds = new ArrayList<Long>();
+        for (LoadBalancerHostMap map : mapDao.findToRemove(LoadBalancerHostMap.class, LoadBalancer.class, lbId)) {
+            hostIds.add(map.getHostId());
+        }
+        return hostIds;
+    }
+
+    @Inject
+    public void setMapDao(GenericMapDao mapDao) {
+        this.mapDao = mapDao;
+    }
+
+    @Inject
+    public void setObjectManager(ObjectManager objectManager) {
+        this.objectManager = objectManager;
+    }
+
+    @Override
+    public Network getManagedNetwork() {
+        return objectManager.findOne(Network.class, NETWORK.NAME, MANAGED_NETWORK_NAME.get());
```

Comment:
We can't hard code this.  We need to get more clever on determining what is the network for the load balancer.  Currently we have one network for everyone.  But going forward in a multi-tenant situation we need to create one "managed" network per account.

Let me think about this one.  Currently what I'm thinking is that we add networkId to the Loadbalancer object.  The UI can select the managed one and pass it in (well, the "UI team" won't like that).  But it seems wrong to tie a LB to a network.  So let's think a bit more on this one.